test/regexmanager.cpp

```diff
@@ -108,13 +108,15 @@ TEST_CASE("RegexManager preserves text when there's nothing to highlight",
 	rxman.quote_and_highlight(input, "feedlist");
 	REQUIRE(input == "xbarx");
 
-	input = "<";
-	rxman.quote_and_highlight(input, "feedlist");
-	REQUIRE(input == "<");
-
 	input = "a<b>";
 	rxman.quote_and_highlight(input, "feedlist");
 	REQUIRE(input == "a<b>");
+
+	SECTION("encode `<` as `<>` for stfl") {
```

Comment:
The original algorithm did not do this.
It is indeed just a consequence of how the new algorithm works.
I do however think it makes sense to do this as stfl wants `<` to be encoded as `<>`.

I think the code calling `quote_and_highlight()` should make sure the string is properly formatted but that is not always the case yet (I'm thinking about fixing that next).

Reply:
Got it!

> I think the code calling quote_and_highlight() should make sure the string is properly formatted but that is not always the case yet (I'm thinking about fixing that next).

The "quote" in the name always confused me, for that same reason. This might be a good place to introduce a new string type (`StflString`?), to statically check that quoted and unquoted strings aren't mixed.

Reply:
Good idea although I prefer to do that in a separate PR.

Reply:
Oh, yeah, certainly; I wasn't suggesting it for this PR.